File: androidApp/src/main/kotlin/fr/abknative/outgo/android/components/ExpenseFilterSelector.kt
package fr.abknative.outgo.android.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.FilterChip
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import fr.abknative.outgo.android.ui.DashboardLabels
import fr.abknative.outgo.android.ui.states.OutgoingFilter

@Composable
fun ExpenseFilterSelector(
    selectedFilter: OutgoingFilter,
    onFilterSelected: (OutgoingFilter) -> Unit,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 8.dp),
        horizontalArrangement = Arrangement.spacedBy(8.dp)
    ) {
        FilterChip(
            selected = selectedFilter == OutgoingFilter.ALL,
            onClick = { onFilterSelected(OutgoingFilter.ALL) },
            label = { Text(DashboardLabels.TAB_ALL) }
        )

        FilterChip(
            selected = selectedFilter == OutgoingFilter.PAID,
            onClick = { onFilterSelected(OutgoingFilter.PAID) },
            label = { Text(DashboardLabels.TAB_PAID) }
        )

        FilterChip(
            selected = selectedFilter == OutgoingFilter.REMAINING,
            onClick = { onFilterSelected(OutgoingFilter.REMAINING) },
            label = { Text(DashboardLabels.TAB_REMAINING) }
        )
    }
}

File: iosApp/iosApp/components/dashboard/ExpenseFilterSelector.swift
import SwiftUI
import SharedApp

struct ExpenseFilterSelector: View {
    // --- Propriétés ---
    let selectedFilter: OutgoingFilter
    let onFilterSelected: (OutgoingFilter) -> Void
    
    // --- Environnement ---
    @Environment(\.outgoColors) private var colors
    @Environment(\.spacing) private var spacing
    
    var body: some View {
        HStack(spacing: 0) {
            
            FilterTabItem(
                label: DashboardLabels.shared.TAB_ALL,
                isSelected: selectedFilter == .all,
                onClick: { onFilterSelected(.all) }
            )
            
            Spacer().frame(width: spacing.large)

            FilterTabItem(
                label: DashboardLabels.shared.TAB_PAID,
                isSelected: selectedFilter == .paid,
                onClick: { onFilterSelected(.paid) }
            )
            
            Spacer().frame(width: spacing.large)

            FilterTabItem(
                label: DashboardLabels.shared.TAB_REMAINING,
                isSelected: selectedFilter == .remaining,
                onClick: { onFilterSelected(.remaining) }
            )
        }
        .padding(.horizontal, spacing.large)
        .frame(maxWidth: .infinity)
        .sensoryFeedback(.selection, trigger: selectedFilter)
    }
}

// --- Sous-composant ---
private struct FilterTabItem: View {
    let label: String
    let isSelected: Bool
    let onClick: () -> Void
    
    @Environment(\.outgoColors) private var colors
    @Environment(\.outgoTypography) private var typo
    @Environment(\.spacing) private var spacing
    
    var body: some View {
        Button(action: {
            withAnimation(.spring(response: 0.3, dampingFraction: 0.7)) {
                onClick()
            }
        }) {
            VStack(alignment: .center, spacing: 4) {
                Text(label.uppercased())
                    .font(typo.body)
                    .fontWeight(isSelected ? .medium : .regular)
                    .foregroundColor(isSelected ? colors.primary : colors.textSecondary)
                    .padding(.vertical, spacing.small)
                
                RoundedRectangle(cornerRadius: 50)
                    .fill(isSelected ? colors.primary : Color.clear)
                    .frame(height: 3)
                    .frame(maxWidth: .infinity)
                    .opacity(isSelected ? 1 : 0)
                    .scaleEffect(x: isSelected ? 1 : 0.4, y: 1)
            }
        }
        .buttonStyle(.plain)
        .frame(maxWidth: .infinity)
        .accessibilityAddTraits(isSelected ? .isSelected : [])
    }
}

// --- Preview ---
#Preview {
    struct PreviewWrapper: View {
        @State var filter: OutgoingFilter = .all
        var body: some View {
            ZStack {
                AppTheme.colorScheme(isDark: false).background.ignoresSafeArea()
                
                VStack {
                    ExpenseFilterSelector(
                        selectedFilter: filter,
                        onFilterSelected: { filter = $0 }
                    )
                    Spacer()
                }
                .padding(.top, 50)
            }
            .outgoTheme()
        }
    }
    return PreviewWrapper()
}
